### DISCUSSION TITLE
FAQ : Usecase for `unseen` and daemon mode

### DISCUSSION THREAD

**Discussion by sweenu** [slurdge/goeland#96] [Ideas]
I was wondering, is there a real usecase for the `unseen` filter?
What can be done with it that is not possible with running goeland every X hours and having the `lasthours(X)` filter?

I am asking this because IMO it adds complexity to both the code and to the usage. Makes goeland stateful, adds a file to be tracked, adds a purge command and related options, makes it a bit harder to package as well 

I think this complexity, if not absolutely needed, makes goeland harder to get started with. We should rather have one way of doing things and document it well.

The same goes for `daemon mode`. It would be better to document well for each platform how to use their cron/daemon systems than have more compexity in the code and the program and make it harder for users.

What do you think?

**Comment by slurdge** [comment]
Hi @sweenu,
Excellent question which I think do belong to FAQ.

To answer, let me go back a little in history. In first iterations, it was like you suggest: only a simple stateless data in, data out program. However this approach showed its limitations and therefore well, compromises to this simple architecture had to be made.

## Why not use `lasthours(X)` with a cronjob every `X` hours ?

First of all, real world RSS feeds are unfortunately not always doing their part in having correct dates and times. So as it happens, sometimes you have the same RSS item with a different time.
Secondly, the main problem is that the schedule is not always constant. I do have `goeland` running only a weekdays for example. If I do a `lasthours` filter I would miss entries.
Thirdly, sometimes the cron is in less than in an hour. For high density RSS feed it's not possible.

Therefore this (ultra-simple) unseen filter. You may note that I selected a pure golang db, opposed to a sqlite or other ORM ?

## Why have a daemon mode with cron instead of an external cron ?

This was added by request from an user. `goeland` strives to be as simple as possible to deploy. Having a simple cron inside `goeland` makes it trivial to deploy to any system even without cron support and have it run the same.
The main reason behind the `daemon` mode is that for docker, instead of a running once program, you need a long-time running process. This simplifies a lot the docker image.

I hope this clarifies why we have those two features ! It is not burned in stone, however I do feel that `goeland`  is still simple to manage so I'm not too worried.

**Reply by sweenu** [reply]
Thanks, the explanations make sense :)
